### PR TITLE
feat: Adds `TermwindServiceProvider` to be able to use with `Laravel` test expectations

### LINKED ISSUE #121: Testing an Artisan command using render()
Is it possible to test an Artisan command's console output (via `expectsOutput`, `doesntExpectOutput`) which is using, for example, `render()` function?

Artisan command `foo`:
```php
...
render('bar'); // even with no styling/HTML
...

```

Unit PEST:
```php
it('does not show output foo string', function () {
    $this->artisan('foo')
        ->doesntExpectOutput('bar');
});

```

Apparently, there is no difference between using `render('bar')` or not: the test always passes.
Am I doing anything wrong?

Thank you

### PR DESCRIPTION
**Pull request by xiCO2k**
This PR adds a `ServiceProvider` that calls the `renderUsing` method while testing, that way nothing will be printed to the CLI and also you can take advantage of the assert methods `expectsOutput` and `doesntExpectOutput`.

## Example:

```php
// console.php
use function Termwind\render;

Artisan::command('hello', function () {
    render('<div><div class="bg-red-500">Hello World</div></div>');
});
```

```php
// Test.php

$this->artisan('hello')->expectsOutput('Hello World');
```

Fixes #121.

### CODE REVIEW

File: src/Laravel/TermwindServiceProvider.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+use Termwind\Termwind;
+
+final class TermwindServiceProvider extends ServiceProvider
+{
+    /**
+     * Sets the the renderer used for unit testing.
+     */
+    public function register(): void
+    {
+        if ($this->app->runningUnitTests()) {
```
Comment (nunomaduro) [inline src/Laravel/TermwindServiceProvider.php:17]: Codewise is perfect. Yet, I think you create a DeferableOutput class that should be used like this:

```php
DeferableOutput::from(fn () => app()->make(...
```

What do you think?

File: src/Laravel/TermwindServiceProvider.php [hunk 2]
```diff
@@ -15,7 +16,9 @@ final class TermwindServiceProvider extends ServiceProvider
     public function register(): void
     {
         if ($this->app->runningUnitTests()) {
-            Termwind::renderUsing(new TestingBufferedOutput($this->app));
+            $this->app->resolving(OutputStyle::class, function ($style): void {
+                Termwind::renderUsing($style->getOutput());
+            });
         }
```
Comment (xiCO2k) [inline src/Laravel/TermwindServiceProvider.php:22]: Figure that we can take advantage of the resolving event.

Comment (xiCO2k) [reply src/Laravel/TermwindServiceProvider.php:22]: That way we get the mocked version from the `PendingCommand`

Comment (nunomaduro) [reply src/Laravel/TermwindServiceProvider.php:22]: Perfect, last question: why only do this in tests?

Comment (xiCO2k) [reply src/Laravel/TermwindServiceProvider.php]: Thats right, It works well for all, on Testing it gets the mocked version from `PendingCommand` and from the `php artisan command` it gets the `ConsoleOutput`.

**Review by nunomaduro — APPROVED** [review]

**Pull request merged**